### PR TITLE
Add cow-shed module

### PR DESCRIPTION
**Pull request by yvesfracari**
This PR:

- Adds CoW Shed module
- Refactor contract codegen path

TODO:
- Add official contracts and init code

### CODE REVIEW

File: examples/cra/package.json [hunk 1]
```diff
@@ -3,13 +3,13 @@
...
   "dependencies": {
-    "@cowprotocol/cow-sdk": "^4.0.3",
+    "@cowprotocol/cow-sdk": "../../",
     "@safe-global/api-kit": "^1.3.0",
     "@safe-global/protocol-kit": "^1.2.0",
     "@testing-library/jest-dom": "^5.16.5",
     "@testing-library/react": "^13.4.0",
     "@testing-library/user-event": "^13.5.0",
-    "ethers": "^5.7.2",
+    "ethers": "5.7.2",
```
Comment (ribeirojose) [inline examples/cra/package.json:12]: 🤔 did the minor version break anything in your code?

File: examples/cra/src/pages/cowShedHook/index.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+import { FormEvent, useCallback, useEffect, useState } from 'react'
+import { CowShedHooks } from '@cowprotocol/cow-sdk'
+import { useCurrentChainId } from 'hooks/useCurrentChainId'
+import { useWeb3Info } from 'hooks/useWeb3Info'
+import { JsonContent } from '../../components/jsonContent'
+import { ResultContent } from '../../components/resultContent'
+import { formatBytes32String } from 'ethers/lib/utils.js'
+import { SigningScheme } from '@cowprotocol/contracts'
+
+const DEADLINE = BigInt(1000000)
```
Comment (ribeirojose) [inline examples/cra/src/pages/cowShedHook/index.tsx:11]: just to make it easier to read
```suggestion
const DEADLINE = BigInt(1_000_000)
```

File: src/common/consts.ts [hunk 3]
```diff
@@ -4,6 +4,10 @@ export const BUY_ETH_ADDRESS = '0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE'
 export const EXTENSIBLE_FALLBACK_HANDLER = '0x2f55e8b20D0B9FEFA187AA7d00B6Cbe563605bF5'
 export const COMPOSABLE_COW = '0xfdaFc9d1902f4e0b84f65F49f244b32b31013b74'
 
+// TODO: Substitute these addresses with the official ones once
+export const COW_SHED_FACTORY = '0x62A0352B8684A7d207D1DdD08f7627b6c42B8b9a'
+export const COW_SHED_IMPLEMENTATION = '0x2f55e8b20D0B9FEFA187AA7d00B6Cbe563605bF5'
+
```
Comment (ribeirojose) [inline src/common/consts.ts:10]: would these be the same address for every chain?

**Review by ribeirojose — APPROVED** [review]
👌

File: src/common/consts.ts (hunk 3 above)
Comment (yvesfracari) [reply src/common/consts.ts:10]: We don't know yet but I think that will be.

File: src/cow-shed/CoWShedHooks.spec.ts [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+import { COW_SHED_PROXY_INIT_CODE } from './proxyInitCode'
+import { SigningScheme } from '@cowprotocol/contracts'
+import { formatBytes32String, solidityPack } from 'ethers/lib/utils'
+import * as contracts from './contracts'
+
+// information from mint and dai example of cow-shed repository
+// https://github.com/cowdao-grants/cow-shed/blob/main/examples/mintDaiAndSwap.ts
+const MOCK_COW_SHED_FACTORY = '0xa9efDEf197130B945462163a0B852019BA529a66'
+const MOCK_COW_SHED_IMPLEMENTATION = '0x49AeF2C4005Bf572665b09014A563B5b9E46Df21'
+const MOCK_INIT_CODE =
```
[the remark this answers is not in the document]
Comment (yvesfracari) [reply src/cow-shed/CoWShedHooks.spec.ts:15]: The init code is generated on the contract deployment, tbh I don't understand exactly how it is generated. Just copied the value generated on my deployment and on the cow-shed test.

**Pull request merged**